docs/unit2.mdx

```diff
@@ -0,0 +1,406 @@
+```python
+if x == y:
+    print('x and y are equal')
+else:
+    if x < y:
+        print('x is less than y')
+    else:
+        print('x is greater than y')
+```
+
+The outer conditional contains two branches. The first branch contains a simple statement. The second branch contains another if statement, which has two branches of its own. Those two branches are both simple statements, although they could have been conditional statements as well.
+
+Although the indentation of the statements makes the structure apparent, nested conditionals become difficult to read very quickly! In general, it is a good idea to avoid them when you can.
+
+Logical operators often provide a way to simplify nested conditional statements. For example, we can rewrite the following code using a single conditional:
+```python
+if 0 < x:
+    if x < 10:
+        print('x is a positive single-digit number.')
+```
+
+The `print` statement is executed only if we make it past both conditionals, so we can get the same effect with the `and` operator:
+```python
+if 0 < x and x < 10:
+    print('x is a positive single-digit number.')
+```
+
+## 2.4
+### A list is a sequence
+Like a string, a **list** is a sequence of values. In a string, the values are characters; in a list, they can be any type. The values in a list are called **elements** or sometimes items.
+
+There are several ways to create a new list; the simplest is to enclose the elements in square brackets, `[ ]`:
+```python
+[10, 20, 30, 40]
+['crunchy frog', 'ram bladder', 'lark vomit']
+```
+
+The first example is a list of four integers. The second is a list of three strings. The elements of a list don’t have to be the same type. The following list contains a string, a float, an integer, and (lo!) another list:
+```python
+['spam', 2.0, 5, [10, 20]]
+```
+
+A list within another list is nested.
+
+A list that contains no elements is called an empty list; you can create one with empty brackets, `[]`.
+
+As you might expect, you can assign list values to variables:
+```python
+>>> cheeses = ['Cheddar', 'Edam', 'Gouda']
+>>> numbers = [17, 123]
+>>> empty = []
+>>> print(cheeses, numbers, empty)
+['Cheddar', 'Edam', 'Gouda'] [17, 123] []
+```
+
+### Lists are mutable
+The syntax for accessing the elements of a list is the same as for accessing the characters of a string—the bracket operator. The expression inside the brackets specifies the index. **Remember that the indices start at 0**:
+```python
+>>> cheeses = ['Cheddar', 'Edam', 'Gouda']
+>>> print(cheeses[0]) 
+Cheddar
+>>> print(cheeses[1])
+Edam
+```
+
+Unlike strings, lists are mutable. When the bracket operator appears on the left side of an assignment, it identifies the element of the list that will be assigned.
+```python
+>>> numbers = [17, 123]
+>>> numbers[1] = 5
+>>> print(numbers)
+[17, 5]
+```
+The element at index 1, which used to be `123`, is now `5`.
```

Comment:
The content from here down to the end of the section is different from the associated reading link that I have https://tealsk12.github.io/2nd-semester-introduction-to-computer-science/readings.md.html#associatedreadings/2.4/listsaremutable. There is an image there too. 
Should I be using another link then?

Reply:
No, I reworded it and decided not to use the image, partially out of laziness and partially because I think the description of the image is more confusing than it is helpful.  If we want an image here to show how indexing and list mutability works, I might rather make a fresh one than use what they had in the teals doc.

Reply:
just put up a dark theme compatible image with description, albeit slightly altered.  we can use the teals wording if y'all prefer that though :)

Reply:
Nah, what you have there is awesome! I wasn't sure if I should be looking for a more up-to-date version of this doc the TEALS Site so I just wanted to make sure